tools/autograd/derivatives.yaml

```diff
@@ -160,16 +175,25 @@
 # in Decalarations.yaml
 - name: abs(Tensor self) -> Tensor
   self: grad * self.sgn()
+  result: auto_element_wise
 
 - name: acos(Tensor self) -> Tensor
   self: grad * -((-self * self + 1).rsqrt()).conj()
+  result: auto_element_wise
 
 - name: add.Tensor(Tensor self, Tensor other, *, Scalar alpha=1) -> Tensor
   self: handle_r_to_c(self.scalar_type(), grad)
   other: handle_r_to_c(other.scalar_type(), maybe_multiply(grad, alpha.conj()))
+  result: "self_fw_grad.defined()? (other_fw_grad.defined()? maybe_multiply(self_fw_grad, alpha) + other_fw_grad : maybe_multiply(self_fw_grad, alpha)) : (other_fw_grad.defined()? other_fw_grad : Tensor())"
+
+- name: add_.Tensor(Tensor(a!) self, Tensor other, *, Scalar alpha=1) -> Tensor(a!)
+  self: handle_r_to_c(self.scalar_type(), grad)
+  other: handle_r_to_c(other.scalar_type(), maybe_multiply(grad, alpha.conj()))
+  result: "self_fw_grad.defined()? (other_fw_grad.defined()? self_fw_grad.mul_(alpha).add_(other_fw_grad) : self_fw_grad.mul_(alpha)) : (other_fw_grad.defined()? other_fw_grad : Tensor())"
```

Comment:
Zero tensor doko (no action needed)

Reply:
More seriously, I'm kind of wondering that for the first version of this PR, if you shouldn't have eschewed the undefined tensor optimization, and just materialized zero tensors for all the tangents that weren't initialized. This would have made the formulas a ton simpler all over the place.

Reply:
A simple solution to this that I use for the draft that implements all the remaining formulas is to use a basic Macro to do these checks.

The materialization is really expensive and that's why I avoided it here.

Reply:
Updated to do the materialization for now